private/pkg/testingextended/testingextended.go

```diff
@@ -27,12 +26,10 @@ func SkipIfShort(t *testing.T) {
 	}
 }
 
-// GetTestTimeout returns the value of the go test -timeout flag.
+// GetTestTimeout returns the time remaining until the test times out or 10m if the test is not set to timeout.
 func GetTestTimeout(t *testing.T) time.Duration {
-	if !flag.Parsed() {
-		t.Fatal("unable to read testing timeout flag as flags have not been parsed")
+	if deadline, ok := t.Deadline(); ok {
```

Comment:
Should we also `!deadline.IsZero()` ... is that even possible?

Reply:
I don't think it's possible, but it doesn't hurt anything to add

Reply:
`The ok result is false if the -timeout flag indicates “no timeout” (0).`